If #include'd so early, "snprintf-test.h" renames the system functions as 
well, missing the whole point (to avoid signature differences).

--- origsrc/heimdal-1.2.1/lib/roken/snprintf.c	2008-07-28 07:04:48.000000000 -0500
+++ src/heimdal-1.2.1/lib/roken/snprintf.c	2009-07-07 12:54:32.318590900 -0500
@@ -35,9 +35,6 @@
 #include <config.h>
 RCSID("$Id: snprintf.c 21005 2007-06-08 01:54:35Z lha $");
 #endif
-#if defined(TEST_SNPRINTF)
-#include "snprintf-test.h"
-#endif /* TEST_SNPRINTF */
 #include <stdio.h>
 #include <stdarg.h>
 #include <stdlib.h>
@@ -45,6 +42,9 @@ RCSID("$Id: snprintf.c 21005 2007-06-08 
 #include <ctype.h>
 #include "roken.h"
 #include <assert.h>
+#if defined(TEST_SNPRINTF)
+#include "snprintf-test.h"
+#endif /* TEST_SNPRINTF */
 
 enum format_flags {
     minus_flag     =  1,
